hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/AbstractHessianInput.java

```diff
@@ -334,12 +335,34 @@ abstract public InputStream readInputStream()
     abstract public Object readObject(Class expectedClass)
             throws IOException;
 
+    /**
+     * Reads an arbitrary object from the input stream.
+     *
+     * @param expectedClass the expected class if the protocol doesn't supply it.
+     * @param expectedTypes the runtime type hints, eg: expectedClass equals Map, expectedTypes can
+     * equals String.class, Short.class
+     */
+    public Object readObject(Class expectedClass, Class<?>... expectedTypes)
```

Comment:
the add method will work in Hessian2Input, but in HessianInput not unsupported. 
I think the HessianInput need to change to support the new methods.